# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeParent.h
# Commit: 6e5d426cd313
# Full Hash: 6e5d426cd3137e9032f0dfcdb09fb7b0d39c1fed
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-07-15 21:39:46
# Regressor Bug: 1547351
# File Overlap Count: 1
# Description:
#   Bug 1547351 - Factor out RecvEmptyTransaction's core transaction building r=sotaro
#   
#   This splits out the inner bit of RecvEmptyTransaction to just iterate over
#   the documents once, rather than iterating over them individually. Originally
#   I ran into difficulties with this and then left it on the table, but I think
# ==============================================================================

diff -r c25628ad5a4c -r 6e5d426cd313 gfx/layers/wr/WebRenderBridgeParent.h
--- a/gfx/layers/wr/WebRenderBridgeParent.h	Mon Jul 15 11:21:15 2019 +0000
+++ b/gfx/layers/wr/WebRenderBridgeParent.h	Mon Jul 15 13:20:31 2019 +0000
@@ -20,6 +20,7 @@
 #include "mozilla/layers/UiCompositorControllerParent.h"
 #include "mozilla/layers/WebRenderCompositionRecorder.h"
 #include "mozilla/Maybe.h"
+#include "mozilla/Result.h"
 #include "mozilla/UniquePtr.h"
 #include "mozilla/WeakPtr.h"
 #include "mozilla/webrender/WebRenderTypes.h"
@@ -306,6 +307,10 @@
     }
   }
 
+  bool ProcessEmptyTransactionUpdates(RenderRootUpdates& aUpdates,
+                                      uint32_t aPaintSequenceNumber,
+                                      bool* aScheduleComposite);
+
   bool SetDisplayList(wr::RenderRoot aRenderRoot, const LayoutDeviceRect& aRect,
                       const wr::LayoutSize& aContentSize, ipc::ByteBuf&& aDL,
                       const wr::BuiltDisplayListDescriptor& aDLDesc,
